### ISSUE TITLE
Type checking issues with worker receive with check

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#19150]
**Description:**
$title. Type checking does not happen for the error relevant to check and the enclosing invokable's return type.

_**Applicable for sync send too.**_

**Steps to reproduce:**
```ballerina
import ballerina/io;

const R = "error reason";

type E error<R>;

public function main() {
    worker w1 returns error? {
        if (true) {
            return error("generic error");
        }

        5 -> w2;
    }

    worker w2 returns E? {
        int j = check <- w1; // the check-ed error is of type `error`
    }

    E? res = wait w2;
    
    if (res is error) {
        io:println("res reason: ", res.reason());
    }
}
```

This results in the following at runtime.
```cmd
$ ballerina run foo.bal 
error: generic error 
        at foo:$lambda$0(foo.bal:10)
```

**Affected Versions:**
```cmd
$ ballerina -v
Ballerina 1.0.0
Language specification 2019R3
```

**Related Issues:**
https://github.com/ballerina-platform/ballerina-lang/issues/18162

**Comment by MaryamZi** [comment]
This seems to be a generic issue with `check`.

See https://github.com/ballerina-platform/ballerina-lang/issues/19299.

Need to check once https://github.com/ballerina-platform/ballerina-lang/issues/19299 is fixed.

**Comment by hasithaa** [comment]
Resolved with #19374
